doc/sphinx/src/tasks.rst

```diff
@@ -3,85 +3,81 @@
 Tasks
 =====
 
+Parthenon's tasking infrastructure is how downstream applications describe 
+and execute their work.  Tasks are organized into a hierarchy of objects.
+``TaskCollection``s have one or more ``TaskRegion``s, ``TaskRegion``s have
+one or more ``TaskList``s, and ``TaskList``s can have one or more sublists
+(that are themselves ``TaskList``s).
+
+Task
+----
+
+Though downstream codes never have to interact with the ``Task`` object directly,
+it's useful to describe nonetheless.  A ``Task`` object is essentially a functor
+that stores the necessary data to invoke a downstream code's functions with
+the desired arguments.  Importantly, however, it also stores information that
+relates itself to other tasks, namely the tasks that must be complete before
+it should execute and the tasks that may be available to run after it completes.
+In other words, ``Task``s are nodes in a directed (possibly cyclic) graph, and
```

Comment:
Maybe I'm not using quite the right words.  The graph can have cycles, but at least one node in the cycle acts as a conditional, either continuing the cycle or exiting it depending on some condition.  Is there a better way to describe that?

Reply:
I think it may be that we just have different mental models for the graph. In my mind, the graph is just defined by task dependencies and its structure is static. Each node in the graph is then associated with a status and I look for tasks that are available to do and try to switch their status. In this model, iterative task lists are just groupings of tasks in the graph where a task can return iterate and that sets the status of all tasks in the iterative list to incomplete.